### ISSUE TITLE
Object Validation in the Settings Editor: Not all errors are shown in the UI

### ISSUE THREAD

**Issue by hediet** [microsoft/vscode#129499]
Testing #129448

Violating
> The number of errors the GUI editor displays should be greater than or equal to the number of errors the JSON editor displays.

![image](https://user-images.githubusercontent.com/2931520/127127409-dae3d8cf-20d5-4f07-9703-9b93220987a3.png)

![image](https://user-images.githubusercontent.com/2931520/127127439-aa69f3fc-09f1-4a5f-a9e0-cdc9b7922409.png)

```json
"contributes": {
    "configuration": {
        "title": "example",
        "properties": {
        "my-custom-example-config": {
            "type": "object",
            "patternProperties": {
                "^#": {
                "type": "string",
                "description": "here here"
                }
            },
            "properties": {
                "la": {
                "type": "string",
                "enum": ["a", "b", "c"],
                "required": true
                },
                "lb": {
                "type": "string",
                "enum": ["b", "c", "d"]
                }
            },
            "additionalProperties": {
                "type": "boolean"
            },
            "default": {}
            }
        }
    }
},
```

**Comment by hediet** [comment]
Also, it seems like missing properties are not reported, neither in the UI nor the json editor.

**Comment by rzhao271** [comment]
I noticed the boolean case failing while writing up the TPI. It should be good in the next Insiders.
I wasn't aware of `required`, but it seems to take in a string array. http://json-schema.org/understanding-json-schema/reference/object.html#required-properties.

**Comment by rzhao271** [comment]
Reopening. I see some issues with both.

**Comment by rzhao271** [comment]
1. My Windows didn't have the latest Insiders. I confirmed the bool issue is fixed.
2. A null value issue I spotted is being tracked in https://github.com/microsoft/vscode/issues/129593
3. Required properties behave strangely. I've opened a new issue for that.

Verifier: to verify this bug, see https://github.com/microsoft/vscode/issues/129448.
In the JSON file, add an additional property of type string. The JSON editor should give an error. The GUI settings editor should also give an error.
